src/biokbase/narrative/jobs/job.py

```diff
@@ -309,24 +293,24 @@ def parameters(self):
         if that's None, then it makes a call to EE2.
 
         If no exception is raised, this only returns the list of parameters, NOT the whole
-        object fetched from ee2.get_job_params
+        object fetched from ee2.check_job
         """
         if self.params is not None:
             return self.params
         else:
             try:
-                self.params = clients.get("execution_engine2").get_job_params(
-                    self.job_id
-                )["params"]
+                state = self.query_ee2_state(self.job_id, init=True)
+                self.update_state(state)
```

Comment:
may as well refresh the whole job so we can use the `update_state` method to set the attributes

Reply:
AFAICT from a bit of grepping, they're always set during job initialisation. I left this code in to be conservative and minimise changes.